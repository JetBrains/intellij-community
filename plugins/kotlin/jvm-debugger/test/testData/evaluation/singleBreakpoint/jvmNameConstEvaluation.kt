const val prefix = "prefix_"

@JvmName("${prefix}f1")
fun f1() = 42

var prop: Int = 0
    @JvmName("${prefix}getter")
    get() {
        return field + 1
    }
    @JvmName("${prefix}setter")
    set(value) {
        field = value + 1
    }

fun main() {
    //Breakpoint!
    val x = 1
}

// EXPRESSION: f1()
// RESULT: 42: I

// EXPRESSION: prop = 1; prop
// RESULT: 3: I