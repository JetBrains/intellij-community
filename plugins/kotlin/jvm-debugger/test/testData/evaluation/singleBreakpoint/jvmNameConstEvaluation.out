LineBreakpoint created at jvmNameConstEvaluation.kt:18
Run Java
Connected to the target VM
jvmNameConstEvaluation.kt:18
Compile bytecode for f1()
Compile bytecode for prop = 1; prop
Disconnected from the target VM

Process finished with exit code 0
